libnd4j/include/ops/declarable/platform/mkldnn/softmax.cpp

```diff
@@ -0,0 +1,185 @@
+                std::unordered_map<int, dnnl::memory> args;
+
+                dnnl::stream stream(engine);
+
+                // provide memory buffers and check whether reorder is required
+
+                // input
+                auto x_user_mem = dnnl::memory(x_user_md, engine, x->getBuffer());
+                const bool xReorder = op_prim_desc.src_desc() != x_user_mem.get_desc();
+                auto x_mkl_mem = xReorder ? dnnl::memory(op_prim_desc.src_desc(), engine) : x_user_mem;
+                if (xReorder)
+                    dnnl::reorder(x_user_mem, x_mkl_mem).execute(stream, x_user_mem, x_mkl_mem);
+                args[DNNL_ARG_SRC] = x_mkl_mem;
+
+                // z
+                auto z_user_mem = dnnl::memory(z_user_md, engine, z->getBuffer());
+                const bool zReorder = op_prim_desc.dst_desc() != z_user_mem.get_desc();
+                auto z_mkl_mem = zReorder ? dnnl::memory(op_prim_desc.dst_desc(), engine) : z_user_mem;
+                args[DNNL_ARG_DST] = z_mkl_mem;
+
+                // run calculations
+                dnnl::softmax_forward(op_prim_desc).execute(stream, args);
+
+                // reorder outputs if necessary
+                if (zReorder)
+                    dnnl::reorder(z_mkl_mem, z_user_mem).execute(stream, z_mkl_mem, z_user_mem);
+
+                stream.wait();
+            }
+
+
+            PLATFORM_IMPL(softmax, ENGINE_CPU) {
+
+                auto input = INPUT_VARIABLE(0);
+                auto output = OUTPUT_VARIABLE(0);
+
+                const int rank = input->rankOf();
+                int dim = block.getIArguments()->size() > 0 ? INT_ARG(0) : rank - 1;
+
+                if (dim < 0) {
+                    dim += rank;
+                }
+
+                REQUIRE_TRUE(dim < rank && dim >= 0, 0, "SOFTMAX_MKLDNN OP: the value of input integer parameter (dimension) must be less than input array rank %i, but got dimension = %i instead !", rank, dim);
+
+                REQUIRE_TRUE(rank <= 6, 0, "SOFTMAX_MKLDNN OP: the rank of input must be less or qual 4, but got rank = %i instead !", rank);
+
+                // mkldnnSoftMax
+                softmaxMKLDNN(input, output, dim);
+
+                return Status::OK();
+            }
+
+            PLATFORM_CHECK(softmax, ENGINE_CPU) {
+
+                auto x = INPUT_VARIABLE(0);
+                auto z = OUTPUT_VARIABLE(0);
+
+                const DataType xType = x->dataType();
+                const DataType zType = z->dataType();
+
+                const int xRank = x->rankOf();
+                bool bSupportedRanks = (xRank > 2 && xRank < 7);
+                /*
+                Source     Destination
+                f32 	    f32
+                f16 	    f16
+                */
+                return  block.isUseMKLDNN() && bSupportedRanks && ((xType == DataType::FLOAT32 && zType == DataType::FLOAT32)
+                    || (xType == DataType::BFLOAT16 && zType == DataType::BFLOAT16));
```

Comment:
According to the documentation, FLOAT32 and FLOAT16 only supported. BFLOAT16 isn't mentioned.

Reply:
as discussed via slack lived only FLOAT32